File: app/src/main/java/com/tuhoang/pocketmind/data/models/ChatMessage.java
package com.tuhoang.pocketmind.data.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "chat_messages")
public class ChatMessage {
    @PrimaryKey(autoGenerate = true)
    public int id;

    public String content;
    public boolean isUser;
    public long timestamp;
    public String imageUrl;

    // Required by Firestore
    public ChatMessage() {
    }

    public ChatMessage(String content, boolean isUser, long timestamp) {
        this.content = content;
        this.isUser = isUser;
        this.timestamp = timestamp;
        this.imageUrl = null;
    }
    
    public ChatMessage(String content, boolean isUser, long timestamp, String imageUrl) {
        this.content = content;
        this.isUser = isUser;
        this.timestamp = timestamp;
        this.imageUrl = imageUrl;
    }
}

File: app/src/main/kotlin/com/tuhoang/pocketmind/data/models/ChatMessage.kt
package com.tuhoang.pocketmind.data.models

data class ChatMessage(
    var content: String = "",
    var isUser: Boolean = false,
    var timestamp: Long = 0L,
    var imageUrl: String? = null
) {
    constructor(content: String, isUser: Boolean, timestamp: Long) : this(
        content = content,
        isUser = isUser,
        timestamp = timestamp,
        imageUrl = null
    )
}
